### PR TITLE
feat: Divider support cssVar

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- ref: #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Divider support cssVar |
| 🇨🇳 Chinese | feat: Divider 组件支持 cssVar |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 73c643c</samp>

This pull request enhances the style customization of the `Divider` component by using CSS variables instead of the style provider. It introduces a new `useCSSVar` hook and a `wrapCSSVar` function that register and update the CSS variables based on the component tokens. It also adds a `unit` function and refactors the `prepareComponentToken` function to ensure consistent and customizable CSS units for the `Divider` style.

### <samp>🤖 Generated by Copilot at 73c643c</samp>

*  Introduce `useCSSVar` hook to wrap divider component with CSS variable provider ([link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5R7), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5L38-R42), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5L88-R91), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-051bc90412cfccd0d26708c9eb9662f454ac5a4025033da796f71009347bce60R1-R4))
*  Remove `wrapSSR` function and replace it with `wrapCSSVar` function ([link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5L38-R42), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5L88-R91))
*  Apply `unit` function to numeric token values to ensure they have CSS units ([link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L27-R31), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L47-R49), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L55-R61), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L67-R69), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L73-R78), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L84-R86), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L116-R118), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L123-R125), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L133-R135), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L142-R144), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L156-R158), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L171-R173), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L178-R194))
*  Export `prepareComponentToken` function and pass it to `genCSSVarRegister` function to transform default token values ([link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L178-R194), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-051bc90412cfccd0d26708c9eb9662f454ac5a4025033da796f71009347bce60R1-R4))
*  Move some imports from `style/index.ts` to `index.tsx` ([link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-fad8d9edfb171ce694f460f4376e63930dcd7283793d0c7bb3faeee2d0d619b5L1-R6), [link](https://github.com/ant-design/ant-design/pull/45810/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5R7))

### CODE REVIEW

File: components/divider/style/index.ts [hunk 1]
```diff
@@ -44,44 +46,44 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
...
+      borderBlockStart: `${unit(lineWidth)} solid ${colorSplit}`,
 
       // vertical
       '&-vertical': {
         position: 'relative',
         top: '-0.06em',
         display: 'inline-block',
         height: '0.9em',
-        marginInline: verticalMarginInline,
+        marginInline: unit(verticalMarginInline ?? 0),
```
Comment (MadCcc) [inline components/divider/style/index.ts:57]:
```suggestion
        marginInline: verticalMarginInline,
```

Comment (MadCcc) [reply components/divider/style/index.ts:57]: 其他的没有 px 单位的也不用加

File: components/divider/style/index.ts [hunk 2]
```diff
@@ -92,35 +94,33 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
 
       [`&-horizontal${componentCls}-with-text-left`]: {
         '&::before': {
-          width: `${orientationMargin * 100}%`,
+          width: `calc(${orientationMargin * 100}%)`,
```
Comment (MadCcc) [inline components/divider/style/index.ts:97]:
```suggestion
          width: `calc(${orientationMargin} * 100%)`,
```

File: components/divider/style/index.ts [hunk 3]
```diff
@@ -92,35 +94,33 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
...
         },
-
         '&::after': {
-          width: `${100 - orientationMargin * 100}%`,
+          width: `calc(${100 - orientationMargin * 100}%)`,
```
Comment (MadCcc) [inline components/divider/style/index.ts:100]:
```suggestion
          width: `calc(100% - ${orientationMargin} * 100%)`,
```

File: components/divider/style/index.ts [hunk 4]
```diff
@@ -92,35 +94,33 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
...
         },
       },
 
       [`&-horizontal${componentCls}-with-text-right`]: {
         '&::before': {
-          width: `${100 - orientationMargin * 100}%`,
+          width: `calc(${100 - orientationMargin * 100}%)`,
```
Comment (MadCcc) [inline components/divider/style/index.ts:106]:
```suggestion
          width: `calc(100% - ${orientationMargin} * 100%)`,
```

File: components/divider/style/index.ts [hunk 5]
```diff
@@ -92,35 +94,33 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
...
         },
-
         '&::after': {
-          width: `${orientationMargin * 100}%`,
+          width: `calc(${orientationMargin * 100}%)`,
```
Comment (MadCcc) [inline components/divider/style/index.ts:109]:
```suggestion
          width: `calc(${orientationMargin} * 100%)`,
```

File: components/divider/style/index.ts [hunk 6]
```diff
@@ -92,35 +94,33 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
...
         },
       },
 
       [`${componentCls}-inner-text`]: {
         display: 'inline-block',
         paddingBlock: 0,
-        paddingInline: textPaddingInline,
+        paddingInline: unit(textPaddingInline ?? 0),
```
Comment (MadCcc) [inline components/divider/style/index.ts:116]:
```suggestion
        paddingInline: textPaddingInline,
```

File: components/divider/style/index.ts [hunk 7]
```diff
@@ -139,7 +139,7 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
       [`&-plain${componentCls}-with-text`]: {
         color: token.colorText,
         fontWeight: 'normal',
-        fontSize: token.fontSize,
+        fontSize: unit(token.fontSize),
```
Comment (MadCcc) [inline components/divider/style/index.ts:142]:
```suggestion
        fontSize: token.fontSize,
```

File: components/divider/style/index.ts [hunk 8]
```diff
@@ -153,7 +153,7 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
           },
 
           [`${componentCls}-inner-text`]: {
-            paddingInlineStart: sizePaddingEdgeHorizontal,
+            paddingInlineStart: unit(sizePaddingEdgeHorizontal),
```
Comment (MadCcc) [inline components/divider/style/index.ts:156]:
```suggestion
            paddingInlineStart: sizePaddingEdgeHorizontal,
```

File: components/divider/style/index.ts [hunk 9]
```diff
@@ -168,13 +168,19 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
           },
 
           [`${componentCls}-inner-text`]: {
-            paddingInlineEnd: sizePaddingEdgeHorizontal,
+            paddingInlineEnd: unit(sizePaddingEdgeHorizontal),
```
Comment (MadCcc) [inline components/divider/style/index.ts:171]:
```suggestion
            paddingInlineEnd: sizePaddingEdgeHorizontal,
```

File: components/divider/style/index.ts [hunk 10]
```diff
@@ -44,44 +46,44 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
...
 
       [`&-horizontal${componentCls}-with-text`]: {
         display: 'flex',
         alignItems: 'center',
-        margin: `${token.dividerHorizontalWithTextGutterMargin}px 0`,
+        margin: `${unit(token.dividerHorizontalWithTextGutterMargin)} 0`,
         color: token.colorTextHeading,
         fontWeight: 500,
-        fontSize: token.fontSizeLG,
+        fontSize: unit(token.fontSizeLG),
```
Comment (MadCcc) [inline components/divider/style/index.ts:78]:
```suggestion
        fontSize: token.fontSizeLG,
```

File: components/divider/style/index.ts [hunk 11]
```diff
@@ -130,7 +130,7 @@ const genSharedDividerStyle: GenerateStyle<DividerToken> = (token): CSSObject =>
       },
 
       [`&-vertical${componentCls}-dashed`]: {
-        borderInlineStartWidth: lineWidth,
+        borderInlineStartWidth: unit(lineWidth),
```
Comment (li-jia-nan) [inline components/divider/style/index.ts:133]:
```suggestion
        borderInlineStartWidth: lineWidth,
```

File: components/divider/style/cssVar.ts [hunk 12]
```diff
@@ -0,0 +1,4 @@
+import { prepareComponentToken } from '.';
+import { genCSSVarRegister } from '../../theme/internal';
+
+export default genCSSVarRegister<'Divider'>('Divider', prepareComponentToken);
```
Comment (MadCcc) [inline components/divider/style/cssVar.ts:4]: 这里得给 orientationMargin 去一下单位，参考 Select

Comment (MadCcc) [reply components/divider/style/cssVar.ts:4]: 确认一下开启关闭 cssvar 样式都是一样的

Comment (li-jia-nan) [reply components/divider/style/cssVar.ts]: done

**Comment by li-jia-nan** [comment]
已经配置了 unitless，不知道为啥没生效

**Review by MadCcc — APPROVED** [review]

**Pull request merged**